src/vs/workbench/parts/search/electron-browser/search.contribution.ts

```diff
@@ -342,34 +342,44 @@ const FocusSearchListCommand: ICommandAction = {
 };
 MenuRegistry.addCommand(FocusSearchListCommand);
 
+const searchInFolderCommand: ICommandHandler = (accessor, resource?: URI) => {
+	const listService = accessor.get(IListService);
+	const viewletService = accessor.get(IViewletService);
+	const panelService = accessor.get(IPanelService);
+	const fileService = accessor.get(IFileService);
+	const resources = getMultiSelectedResources(resource, listService, accessor.get(IEditorService));
+
+	return openSearchView(viewletService, panelService, true).then(searchView => {
+		if (resources && resources.length) {
+			return fileService.resolveFiles(resources.map(resource => ({ resource }))).then(results => {
+				const folders: URI[] = [];
+
+				results.forEach(result => {
+					if (result.success) {
+						folders.push(result.stat.isDirectory ? result.stat.resource : dirname(result.stat.resource));
+					}
+				});
+
+				searchView.searchInFolders(distinct(folders, folder => folder.toString()), (from, to) => relative(from, to));
+			});
+		}
+
+		return void 0;
+	});
+};
+
 const FIND_IN_FOLDER_ID = 'filesExplorer.findInFolder';
-CommandsRegistry.registerCommand({
+KeybindingsRegistry.registerCommandAndKeybindingRule({
 	id: FIND_IN_FOLDER_ID,
-	handler: (accessor, resource?: URI) => {
-		const listService = accessor.get(IListService);
-		const viewletService = accessor.get(IViewletService);
-		const panelService = accessor.get(IPanelService);
-		const fileService = accessor.get(IFileService);
-		const resources = getMultiSelectedResources(resource, listService, accessor.get(IEditorService));
-
-		return openSearchView(viewletService, panelService, true).then(searchView => {
-			if (resources && resources.length) {
-				return fileService.resolveFiles(resources.map(resource => ({ resource }))).then(results => {
-					const folders: URI[] = [];
-
-					results.forEach(result => {
-						if (result.success) {
-							folders.push(result.stat.isDirectory ? result.stat.resource : dirname(result.stat.resource));
-						}
-					});
-
-					searchView.searchInFolders(distinct(folders, folder => folder.toString()), (from, to) => relative(from, to));
-				});
-			}
+	weight: KeybindingWeight.WorkbenchContrib,
+	when: ContextKeyExpr.and(ExplorerFolderContext, ResourceContextKey.Scheme.isEqualTo(Schemas.file)), // todo@remote
+	primary: KeyMod.Shift | KeyMod.Alt | KeyCode.KEY_F,
+	handler: searchInFolderCommand
+});
 
-			return void 0;
-		});
-	}
+CommandsRegistry.registerCommand({
```

Comment:
I think this is registering the command twice now. `registerCommandAndKeybindingRule` should do this work already